omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,242 @@
+			errors.add("source");
+		} else {
+			identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+			if (identifierSource == null) {
+				errors.add("source");
+			}
+		}
+		if (identifierTypeUuid == null || StringUtils.isBlank(identifierTypeUuid.toString())) {
+			errors.add("identifierType");
+		} else {
+			patientIdentifierType = Context.getPatientService()
+			        .getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+			if (patientIdentifierType == null) {
+				errors.add("IdentifierType");
+			}
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException(
+			        "The values of the following inputs are missing or invalid : " + errors.toString());
+		}
+		Location location = null;
+		if (locationUuid != null && StringUtils.isNotBlank(locationUuid.toString())) {
+			location = Context.getLocationService().getLocationByUuid(locationUuid.toString());
+		}
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		autoGenerationOption.setSource(identifierSource);
+		autoGenerationOption.setIdentifierType(patientIdentifierType);
+		autoGenerationOption.setManualEntryEnabled(this.parseBoolean(manualEntryEnabled));
+		autoGenerationOption.setAutomaticGenerationEnabled(this.parseBoolean(automaticallyEnabled));
+		autoGenerationOption.setLocation(location);
+		return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.FULL);
+	}
+	
+	@Override
+	public SimpleObject update(String uuid, SimpleObject postBody, RequestContext context) throws ResponseException {
+		int id = 0;
+		try {
+			id = Integer.parseInt(uuid);
+		}
+		catch (NumberFormatException e) {
+			throw new ValidationException("The uuid " + e.getMessage() + "can not be converted to an integer");
+		}
+		AutoGenerationOption autoGenerationOption = identifierSourceService.getAutoGenerationOption(id);
+		if (autoGenerationOption == null) {
+			throw new ObjectNotFoundException();
+		}
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticGenerationEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		IdentifierSource identifierSource = null;
+		if (sourceUuid != null && StringUtils.isNotBlank(sourceUuid.toString())) {
+			identifierSource =  identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+		}
+		Location location = null;
+		if (locationUuid != null && StringUtils.isNotBlank(locationUuid.toString())) {
+			location = Context.getLocationService().getLocationByUuid(locationUuid.toString());
+		}
+		if (location == null && identifierSource == null && automaticGenerationEnabled == null
+		        && manualEntryEnabled == null) {
+			throw new ResourceDoesNotSupportOperationException(
+			        "updating an autoGenerationOption with missing parameters is not supported");
```

Comment:
In the error message, can you mention the missing parameters? Also ensure that the message does not appear like they are all mandatory.